department-of-veterans-affairs/caseflow
# Ensure email notification history displays updated emails

Resolves #14988

### Description
Dogfooding central to virtual flow revealed that the email notification history was not displaying the emails sent when the background job completes and alerts are updated. This PR resolves this issue by refreshing the email events that returned from the endpoint `/virtual_hearing_job_status`.

### Acceptance Criteria
- [x] Ensure the backend is returning new email events after the create conference job is finishing running

### Testing Plan
This is tricky to test as the `CreateConferenceJob` runs synchronously in dev as opposed to PROD where it runs in the background. In order to test, we need stall the polling from completing. 
1. Replace this [line](https://github.com/department-of-veterans-affairs/caseflow/blob/2cd334c11bd12c6fc62ce9be6d4b7ce02e411d46/app/models/hearings/forms/base_hearing_update_form.rb#L130) with the following; this will prevent the polling from finishing
> VirtualHearings::CreateConferenceJob.perform_later(job_args)
2. Enable feature toggle on rails console`FeatureToggle.enable!(:full_page_video_to_virtual, users: [User.find_by(css_id: "BVASYELLOW")])` and `FeatureToggle.enable!(:schedule_virtual_hearings_for_central, users: [User.find_by(css_id: "BVASYELLOW")])` 

3. Login as `BVASYELLOW`
4. Go to http://localhost:3000/hearings/schedule and pick a video or central hearing day
Pick any hearings to convert to virtual hearing 
5. Convert the hearing to `Virtual` 
6. See blue progress alert on top and scroll down to see email notification history does not display. In rails console do the following(This will ensure that polling will finish successfully) 
```
hearing = Hearing.find_by(uuid: PASTE_UUID_HERE) or LegacyHearing.find_by(vacols_id: PASTE_VACOLS_ID_HERE) 
VirtualHearings::CreateConferenceJob.perform_now(hearing_id: hearing.id, hearing_type: hearing.class.name)
``` 
7. See that blue alert changes to a green alert and email notification history is present
